common.mk

```diff
@@ -142,15 +141,34 @@ FIRTOOL_TARGETS = \
 	$(FIRTOOL_SMEMS_CONF) \
 	$(FIRTOOL_TOP_SMEMS_JSON) \
 	$(FIRTOOL_TOP_HRCHY_JSON) \
-	$(FIRTOOL_MODEL_MOD_HRCHY_JSON) \
+	$(FIRTOOL_MODEL_HRCHY_JSON) \
 	$(FIRTOOL_MODEL_SMEMS_JSON) \
 	$(FIRTOOL_FILELIST) \
 	$(FIRTOOL_BB_MODS_FILELIST)
 
+ifeq (,$(ENABLE_CUSTOM_FIRRTL_PASS))
+	REPL_SEQ_MEM = none
```

Comment:
I'm not sure I like this `none` usage. It should be clear that certain recipes generate certain files (and not have a layer of abstraction with "dummy" files called `none`). Maybe instead of having 2 steps (generate the collateral needed to generate the `TOP/HARNESS_SMEMS_CONF`, then actually generate it) instead just combine https://github.com/ucb-bar/chipyard/blob/092a4a3b894684106d7b7d95772eb590e95815c6/common.mk#L198-L204 into the https://github.com/ucb-bar/chipyard/blob/092a4a3b894684106d7b7d95772eb590e95815c6/common.mk#L150 step. Then your targets would just be generated in 1 step:

```make
FIRRTL_COMPILER_TARGETS = \
	$(FIRTOOL_TOP_HRCHY_JSON) \
	$(FIRTOOL_MODEL_MOD_HRCHY_JSON) \
	$(FIRTOOL_FILELIST) \
	$(FIRTOOL_BB_MODS_FILELIST) \
        $(TOP_SMEMS_CONF) \
        $(HARNESS_SMEMS_CONF)
```

Reply:
I tried merging these two, but it started to break for the `MidFIRRTL` to `Verilog` case. I think this is because the `SFC` pass that strips out SRAMs and replaces them with blackboxes are performed in `MidFIRRTL` & if `REPL_SEQ_MEM` is not `none`, the SFC will perform this pass for designs with `Fixed` inside them. Hence, `FIRTOOL` does not generate the `FIRTOOL_SMEMS_CONF` file because the `SRAM` pass has been already executed by `SFC`. [](https://cs.github.com/chipsalliance/firrtl/blob/135739f0c7ef3f6404ba7115f77c7d7b913f6748/src/main/scala/firrtl/passes/memlib/ReplaceMemTransform.scala#L118) this link seems to show that the pass is performed in `MidFIRRTL`. 
Admittedly, the current version is ugly & I would also like to come up with a better method. Any suggestions?

Reply:
So if I understand correctly, this can be fixed by ignoring the `MidFIRRTL` case and instead having 2 flows (SFC NoOp -> All Firtool, SFC to LowFIRRTL -> Firtool LowFIRTOOL to Vlog)?